app/services/create_account.rb

```diff
@@ -13,9 +13,12 @@ def save
     account.save && create_external_resources ? true : false
   end
 
+  # `Apartment::Tenant.create` calls the DB adapter's `switch`, which we have a hook into
+  # via an initializer.  In our hook we do `account.switch!` and that requires a well-formed
+  # Account (i.e. creation steps complete, endpoints populated).  THEREFORE, `create_tenant`
+  # must be called *after* all external resources are provisioned.
   def create_external_resources
-    create_tenant &&
-      create_account_inline
+    create_account_inline && create_tenant
```

Comment:
UUID exists as part of the Account model.  It is not coined by tenant schema creation.